TROUBLESHOOTING.md

```diff
@@ -119,13 +119,24 @@ It will start profiling when you start the application, then record the profilin
 Some example usages are:
 
 ```shell script
+PATH_TO_ASYNC_PROFILER=...
+
 # profile CPU startup
-java -agentpath:/path/to/async-profiler/build/libasyncProfiler.so=start,event=cpu,file=startup-cpu-profile.html,interval=1000000,simple\
-    -jar my-application.jar
+java -agentpath:${PATH_TO_ASYNC_PROFILER}/lib/libasyncProfiler.so=start,event=cpu,file=startup-cpu-profile.html,interval=1000000,simple\
+    -jar target/quarkus-app/quarkus-run.jar
 
 # profile allocation startup
-java -agentpath:/path/to/async-profiler/build/libasyncProfiler.so=start,event=alloc,file=/tmp/startup-alloc-profile.html,interval=1000000,simple\
-    -jar my-application.jar
+java -agentpath:${PATH_TO_ASYNC_PROFILER}/lib/libasyncProfiler.so=start,alloc=1,total,event=alloc,file=startup-alloc-profile.jfr -XX:+UnlockExperimentalVMOptions -XX:+UseEpsilonGC -XX:-UseTLAB -Xmx1G -Xms1G -XX:+AlwaysPreTouch -jar target/quarkus-app/quarkus-run.jar
```

Comment:
Thanks for the comment but I have some further comments:

> Don't need for -XX:+AlwaysPreTouch and both -XX:+UseEpsilonGC or -XX:+UseG1GC - with the latter which improve tremendously the life of users, because doesn't risk to OOM because cannot recycle memory.

I don't understand what you're saying tbh. This command line was apparently given by you to Roberto. I'm just using it there.

Can you please provide an alternative command if you think it's not good?

Keep in mind that this command is used solely to profile the startup. So I'm not sure what `improve tremendously the life of users` have to do with it? Here we want to get the best picture of the allocations. That's all we want.

> I suggest to use a duration, which is realistic enough considered the actual startup time with the tracing enabled (which can be of many seconds i.e. +12 seconds too!)

I don't think that's a good option for this particular purpose. I think we want the control of when we stop the app.
The shutdown of a Quarkus application can't be mixed really, it's very well isolated in the flamegraph so I would rather use something that is deterministic and you can stop when you want, rather than something with a timeout that might not be very practical.

Reply:
Let's not let perfection be the enemy of progress on this. I think that we all agree that what @gsmet has added here is an improvement over the what exists in the guide, so best get it in and @franz1981 or anyone else can further improve when they have the time.

Reply:
Agree on this @geoand but I'm just brain dumping before I forgot. If the changes here are not possible, I'll send myself a PR

The point about `-XX:+AlwaysPreTouch` is that was used for a different purpose (i.e. paying upfront on the RSS side the full cost of a used heap) - and the `-XX:+UseEpsilonGC` requires users to carefully size the heap or they will get OOM while profiling, which is a PITA.

I hope that explain why I've suggested to remove the first and and switch to G1 (no need for experimental options too, at this poin!) - while keeping `-XX:-UseTLAB`.

So, TLDR @gsmet 
> Can you please provide an alternative command if you think it's not good?
Yep, is:
```
java -agentpath:${PATH_TO_ASYNC_PROFILER}/lib/libasyncProfiler.so=start,alloc=1,total,event=alloc=1,file=startup-alloc-profile.jfr -XX:+UseG1GC -XX:-UseTLAB -Xmx1G -Xms1G -jar target/quarkus-app/quarkus-run.jar
```

> I don't think that's a good option for this particular purpose. I think we want the control of when we stop the app.

In that case, your choice: I personally prefer to stop the profiler manually i.e. 
```${PATH_TO_ASYNC_PROFILER}/bin/asprof stop -f startup-alloc-profile.jfr <pid of the java app>```

But if you prefer instead to have the stop, as you rightly said, it should be clearly visible in the flamegraph, and could just be ignored.